### DISCUSSION TITLE
Two handlers on a message: common and special

### DISCUSSION THREAD

**Discussion by survtur** [aiogram/aiogram#600] [Q&A] [answered]
I would like to make some prehandler to process on every incoming message before it gets to its personal handler. This prehandler will just read message, not change it. What is the best way to do it? 

I don't want to manually add my processing fucntion to the beginning of every handler.
There should be some kind of  handler which is "pass-through" like in python logging.

**Comment by Olegt0rr** [comment, marked as answer]
Look at the middlewares:
https://github.com/aiogram/aiogram/blob/dev-2.x/examples/middleware_and_antiflood.py

**Reply by survtur** [reply]
Yes, thank. It solves my problem.

I wonder what would be solution to do something with all messages bot sends?

**Reply by Olegt0rr** [reply]
Output middleware will be available with aiogram 3.0
